rmgpy/molecule/generator.py

```diff
@@ -501,24 +501,33 @@ def generate_minimum_resonance_isomer(mol):
     """
 
     cython.declare(
+        atom=Atom,
         candidates=list,
         sel=Molecule,
         cand=Molecule,
         metric_sel=list,
+        charge_sel=int,
+        charge_cand=int,
         metric_cand=list,
         )
 
-
     candidates = resonance.generateIsomorphicResonanceStructures(mol)
-    
-    sel = candidates[0]
-    metric_sel = get_unpaired_electrons(sel)
-    for cand in candidates[1:]:
-       metric_cand = get_unpaired_electrons(cand)
-       if metric_cand < metric_sel:
-            sel = cand
-            metric_sel = metric_cand
 
+    sel = mol
+    metric_sel = get_unpaired_electrons(sel)
+    charge_sel = 0
+    for atom in sel.vertices:
+        charge_sel += abs(atom.charge)
+    for cand in candidates:
+        metric_cand = get_unpaired_electrons(cand)
+        if metric_cand < metric_sel:
+            charge_cand = 0
+            for atom in cand.vertices:
+                charge_cand += abs(atom.charge)
+            if charge_cand < charge_sel:
```

Comment:
Perhaps this should be `<=` so that the selected structure will still be updated if the indices are reduced but the charge remains the same.